### PR TITLE
[DX-352] Update upgrading-tyk docs

### PR DESCRIPTION
**Pull request by letzya**
https://deploy-preview-2425--tyk-docs.netlify.app/docs/nightly/upgrading-tyk/

## Types of changes

- [ ] Fixing typo (please merge to production)
- [ ] Documenting a new feature (please merge to production)
- [ ] Documentation for future release (please do not merge to production)
- [ ] Something else (please add if needs merging to production or not)

## Checklist

- [ ] I have [reviewed the guidelines](../CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](../CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I have added a preview link.

### CODE REVIEW

File: tyk-docs/content/upgrading-tyk.md [hunk 1]
```diff
@@ -7,13 +7,27 @@ menu:
 ---
 
 ## Introduction
-Follow the instructions relevant to your Tyk setup to upgrade your Tyk components.
-Note: Upgrading Tyk will not overwrite your configuration files.  However, it is especially good practice to routinely back these files up, especially right before upgrading your software.
 
-## Tyk Cloud Classic
-Tyk Cloud Classic users are automatically upgraded to the latest version as soon as it's released.
+Please read this through before upgrading any of Tyk components.
```
Comment (andyo-tyk) [inline tyk-docs/content/upgrading-tyk.md]:
```suggestion
Please read this through before upgrading any of your Tyk components.
```

File: tyk-docs/content/upgrading-tyk.md [hunk 2]
```diff
@@ -7,13 +7,27 @@ menu:
...
 
-## Tyk Multi-Cloud Gateway
+This page aims to provide guidance for upgrading Tyk products. When upgrading Tyk, you'd need to upgrade separately every component that you are using. To avoid confusion each product offering and within it each component will have a separate section discussing and detailing the relevant information.
```
Comment (andyo-tyk) [inline tyk-docs/content/upgrading-tyk.md]:
```suggestion
This page provides guidance for upgrading your Tyk installation. When upgrading Tyk, you need to consider every component (e.g. Gateway, Pump, Dashboard) separately taking into consideration the style of deployment you've implemented. We have structured this guide by deployment type (e.g. Self-Managed, Multi-DC) to keep all the information you need in one place.
```

File: tyk-docs/content/upgrading-tyk.md [hunk 3]
```diff
@@ -7,13 +7,27 @@ menu:
...
+
+All our components share a few common standards:
+- We do not introduce breaking changes unless specifically stated in the release (and it rarely happens).
```
Comment (andyo-tyk) [inline tyk-docs/content/upgrading-tyk.md]:
```suggestion
- We do not introduce breaking changes unless specifically stated in the release notes (and it rarely happens).
```

File: tyk-docs/content/upgrading-tyk.md [hunk 4]
```diff
@@ -7,13 +7,27 @@ menu:
...
+- Upgrade does not overwrite your configuration files, however, it is a good practice to routinely back these files up (using a git or another tool), especially right before upgrading your software.
```
Comment (andyo-tyk) [inline tyk-docs/content/upgrading-tyk.md]:
```suggestion
- Upgrade does not overwrite your configuration files, however, it is good practice to back these files up routinely (using git or another tool); we strongly recommend you take a backup before upgrading Tyk.
```

File: tyk-docs/content/upgrading-tyk.md [hunk 5]
```diff
@@ -7,13 +7,27 @@ menu:
...
+- Upgradng essentially means pulling the new images from https://hub.docker.com/u/tykio, and linux packages from https://packagecloud.io/tyk/
```
Comment (andyo-tyk) [inline tyk-docs/content/upgrading-tyk.md]:
```suggestion
- Upgrading essentially means pulling the new images from https://hub.docker.com/u/tykio, and linux packages from https://packagecloud.io/tyk/ 
```

File: tyk-docs/content/upgrading-tyk.md [hunk 6]
```diff
@@ -7,13 +7,27 @@ menu:
...
+Please read this through before upgrading any of Tyk components.
 
-## Tyk Multi-Cloud Gateway
+This page aims to provide guidance for upgrading Tyk products. When upgrading Tyk, you'd need to upgrade separately every component that you are using. To avoid confusion each product offering and within it each component will have a separate section discussing and detailing the relevant information.
+
+All our components share a few common standards:
+- We do not introduce breaking changes unless specifically stated in the release (and it rarely happens).
+- Upgrade does not overwrite your configuration files, however, it is a good practice to routinely back these files up (using a git or another tool), especially right before upgrading your software.
+- Upgradng essentially means pulling the new images from https://hub.docker.com/u/tykio, and linux packages from https://packagecloud.io/tyk/ 
+- You do not need to migrate or run migration scripts unless specifically stated in the release (and it rarely happens).
```
Comment (andyo-tyk) [inline tyk-docs/content/upgrading-tyk.md]:
```suggestion
- You do not need to migrate or run migration scripts for your APIs, policies or other assets created in Tyk unless specifically stated in the release (and it rarely happens).
```

File: tyk-docs/content/upgrading-tyk.md [hunk 7]
```diff
@@ -7,13 +7,27 @@ menu:
...
+- Check our [versioning and long-term-support policies]({{< ref "frequently-asked-questions/long-term-support-releases/" >}}) for more details on the way we release miajor, minor, new features, patches and the supported releases.
```
Comment (andyo-tyk) [inline tyk-docs/content/upgrading-tyk.md]:
```suggestion
- Check our [versioning and long-term-support policies]({{< ref "frequently-asked-questions/long-term-support-releases/" >}}) for more details on the way we release major and minor features, patches and how long we support each release.
```

File: tyk-docs/content/upgrading-tyk.md [hunk 8]
```diff
@@ -7,13 +7,27 @@ menu:
...
+- If you experience any issues with the new version you pulled, please contact Tyk Support for assistance support@tyk.io
```
Comment (andyo-tyk) [inline tyk-docs/content/upgrading-tyk.md]:
```suggestion
- If you experience any issues with the new version you pulled, please contact Tyk Support for assistance at support@tyk.io
```

File: tyk-docs/content/upgrading-tyk.md [hunk 9]
```diff
@@ -7,13 +7,27 @@ menu:
...
+
+## Tyk OSS Gateway
+
+### docker / Kubernetes
+To upgrade the gateway simple pull/use the image you want to use from [docker hub](https://hub.docker.com/r/tykio/tyk-gateway/tags)
```
Comment (andyo-tyk) [inline tyk-docs/content/upgrading-tyk.md]:
```suggestion
To upgrade the Gateway simply pull/use the image you want to use from [docker hub](https://hub.docker.com/r/tykio/tyk-gateway/tags) 
```

**Review by andyo-tyk — DISMISSED** [review]

File: tyk-docs/content/upgrading-tyk.md [hunk 10]
```diff
@@ -7,22 +7,150 @@ menu:
...
+## Tyk OSS Gateway
+
+### List of releases
+
+To get the list of image releases, for containerised deployments such as Docker and Kubernetes use [docker hub](https://hub.docker.com/r/tykio/tyk-gateway/tags)
+
+### Docker
+
+#### Simple environment or a Dev environment
+In a simple environment or a development environment, in which you can simple restart your gateways, do the follwing:
```
Comment (dcs3spp) [inline tyk-docs/content/upgrading-tyk.md]:
```suggestion
In a development environment, in which you can simply restart your gateways, do the following:
```

File: tyk-docs/content/upgrading-tyk.md [hunk 11]
```diff
@@ -7,22 +7,150 @@ menu:
...
+
+To get the list of image releases, for containerised deployments such as Docker and Kubernetes use [docker hub](https://hub.docker.com/r/tykio/tyk-gateway/tags)
+
+### Docker
+
+#### Simple environment or a Dev environment
+In a development environment, in which you can simply restart your gateways, do the following:
+
+1. Backup your gateway config file
+2. Update the image version in the docker dommand or script
```
Comment (dcs3spp) [inline tyk-docs/content/upgrading-tyk.md]:
```suggestion
2. Update the image version in the docker command or script
```

File: tyk-docs/content/upgrading-tyk.md [hunk 12]
```diff
@@ -7,22 +7,150 @@ menu:
...
+- You do not need to migrate or run migration scripts for your APIs, policies or other assets created in Tyk unless specifically stated in the release (and it rarely happens).
+- Check our [versioning and long-term-support policies]({{< ref "frequently-asked-questions/long-term-support-releases/" >}}) for more details on the way we release major and minor features, patches and how long we support each release.
+- If you experience any issues with the new version you pulled, please contact Tyk Support for assistance at support@tyk.io
 
-### For MacOS Users
+## Tyk OSS Gateway
+
+### List of releases
+
+To get the list of image releases, for containerised deployments such as Docker and Kubernetes use [docker hub](https://hub.docker.com/r/tykio/tyk-gateway/tags)
```
Comment (dcs3spp) [inline tyk-docs/content/upgrading-tyk.md]:
```suggestion
To get the list of image releases for containerised deployments such as Docker and Kubernetes, please visit [docker hub](https://hub.docker.com/r/tykio/tyk-gateway/tags)
```

File: tyk-docs/content/upgrading-tyk.md [hunk 13]
```diff
@@ -7,22 +7,150 @@ menu:
...
+- You do not need to migrate or run migration scripts for your APIs, policies or other assets created in Tyk unless specifically stated in the release (and it rarely happens).
+- Check our [versioning and long-term-support policies]({{< ref "frequently-asked-questions/long-term-support-releases/" >}}) for more details on the way we release major and minor features, patches and how long we support each release.
+- If you experience any issues with the new version you pulled, please contact Tyk Support for assistance at support@tyk.io
 
-### For MacOS Users
+## Tyk OSS Gateway
+
+### List of releases
+
+To get the list of image releases for containerised deployments such as Docker and Kubernetes, please visit [docker hub](https://hub.docker.com/r/tykio/tyk-gateway/tags)
```
Comment (dcs3spp) [inline tyk-docs/content/upgrading-tyk.md]:
```suggestion
To get the list of image releases for containerised deployments, such as Docker and Kubernetes, please visit [docker hub](https://hub.docker.com/r/tykio/tyk-gateway/tags)
```

File: tyk-docs/content/upgrading-tyk.md [hunk 14]
```diff
@@ -7,22 +7,150 @@ menu:
...
-### For MacOS Users
+## Tyk OSS Gateway
+
+### List of releases
+
+To get the list of image releases for containerised deployments, such as Docker and Kubernetes, please visit [docker hub](https://hub.docker.com/r/tykio/tyk-gateway/tags)
+
+### Docker
+
+#### Simple environment or a Dev environment
```
Comment (dcs3spp) [inline tyk-docs/content/upgrading-tyk.md]:
```suggestion
#### Development environment
```

File: tyk-docs/content/upgrading-tyk.md [hunk 15]
```diff
@@ -7,22 +7,150 @@ menu:
...
+
+### Docker
+
+#### Development environment
+In a development environment, in which you can simply restart your gateways, do the following:
+
+1. Backup your gateway config file
+2. Update the image version in the docker command or script
+3. Restart the gateway. For example, update the follwing command to `v5.0` and run it as follows:
+```bash
```
Comment (dcs3spp) [inline tyk-docs/content/upgrading-tyk.md]:
```suggestion

```bash
$ docker run \
  --name tyk_gateway \
  --network tyk \
  -p 8080:8080 \
  -v $(pwd)/tyk.standalone.conf:/opt/tyk-gateway/tyk.conf \
  -v $(pwd)/apps:/opt/tyk-gateway/apps \
  docker.tyk.io/tyk-gateway/tyk-gateway:v5.0
```
```

Comment (letzya) [reply tyk-docs/content/upgrading-tyk.md]: @dcs3spp sorry , can't understand the suggestion

File: tyk-docs/content/upgrading-tyk.md [hunk 16]
```diff
@@ -1,66 +1,197 @@
...
+   <br>
+   For example, this [docker-compose.yaml](https://github.com/TykTechnologies/tyk-gateway-docker/blob/e44c765f4aca9aad2a80309c5249ff46b308e46e/docker-compose.yml#L4) has this line `image: docker.tyk.io/tyk-gateway/tyk-gateway:v4.3.3`. Change `4.3.3` to the version you want to use.
+3. Restart the gateway (or stop and start it)
+```console
+docker compose restart 
 ```
-### For Linux Users
-```{.copyWrapper}
-wget https://raw.githubusercontent.com/lonelycode/tyk-hybrid-docker/master/start.sh
-chmod +x start.sh
```
Comment (letzya) [inline tyk-docs/content/upgrading-tyk.md:34]: @OldStubbsy I removed the reference to start.sh, is that okay?

File: tyk-docs/content/upgrading-tyk.md [hunk 17]
```diff
@@ -69,46 +200,40 @@ In a production environment, where we recommend installing the Dashboard, Gatewa
 
 Tyk is compatible with a blue-green or rolling update strategy.
 
-For a single machine installation, you should follow the instructions below for your operating system.
-
-Our repositories will be updated at [https://packagecloud.io/tyk](https://packagecloud.io/tyk) when new versions are released. As you set up these repositories when installing Tyk to upgrade all Tyk components  you can run:
+For a single-machine installation, you should follow the instructions below for your operating system.
 
-### For Ubuntu
+### Ubuntu Upgrade
```
Comment (letzya) [inline tyk-docs/content/upgrading-tyk.md:211]: @OldStubbsy Can you please help with a command for this section

**Review by sedkis — APPROVED** [review]

**Review by OldStubbsy — APPROVED** [review]
Super!

**Pull request merged**